Refactor/Rewrite

mattheu opened the pull request:
There were a couple of issues with this plugin... 
1. Authentication seemed to expire.
2. Some sites didn't even seem to work properly on some accounts.
3. A very low number of posts was returned each query, resulting in many many API requests on larger sites.
4. I think The 3rd party API class was using an old vesion of the API

I have updated this to use the PHP class provided by Google. It does proper oAuth so should be more robust. Updated the admin to allow the user to save all the API details. 

Also update the API to return 1000 results for each API request so should be much better on larger sites. Note this version of the API supports up to 10,000 results per query.

~~Todo - remove unused files from Google API PHP class.~~ Not going to do this

mattheu committed:
Rewrite. Use google api php client

mattheu committed:
ability to add api details in the admin

mattheu committed:
re-add opt-out

mattheu committed:
tidy up

mattheu committed:
bit of a tidy again.

mattheu committed:
use separate settings field for each field #UI

mattheu committed:
only sleep after to avoid unnecessary delays when only a single query is required

mattheu committed:
delete functions

mattheu committed:
Merge branch 'refactor' of github.com:humanmade/hm-top-posts into refactor

Conflicts:
	hmtp.admin.php

mattheu committed:
re-add opt-out

mattheu committed:
remove sleep - mis-read docs on rate limiting - they handle it

mattheu committed:
use date from args, default to 1 month ago

mattheu committed:
oops can't do that

mattheu committed:
wrong args

mattheu committed:
delete examples

mattheu committed:
ignore .ds_store

mattheu committed:
widget

Commit:
formatting

Commit:
wp_nonce_field already echoes by default

Commit:
Docblocks

mattheu committed:
Merge pull request #6 from humanmade/wp-standards-4

Coding standards and inline docs

mattheu committed:
bug fix - properties can have multipple profiles

mattheu committed:
Merge branch 'refactor' of github.com:humanmade/hm-top-posts into refactor

Conflicts:
	hmtp.admin.php

mattheu committed:
no closures

mattheu committed:
delete old class

joehoyle committed:
Add a filter_callback funciton to the caller can decide what is included in the returned posts

mattheu committed:
comment out API key setting

mattheu committed:
Add cached URL to postID from VIP

mattheu committed:
Allow defining settings

mattheu committed:
remove API key and hide other fields when defined

mattheu committed:
use defined constant to set query start date

mattheu committed:
Allow Filters

mattheu committed:
Formatting

mattheu committed:
remove home_url check

mattheu committed:
Hanlde pages & page on front

mattheu committed:
docs and formatting

mattheu committed:
Fix widget output

mattheu committed:
allow filtering widget output

mattheu committed:
constants names should be quoted when checking if defined

mattheu committed:
fix use of $this in a closure breaking all things

mattheu committed:
Update HM Top posts Strings

joehoyle committed:
Increment the index in the options, otherwise it doesn't do anything

Commit:
Build an absolute URL as the Google Analytics API returns a relative URL

Commit:
Move the view count outside the link

Commit:
Return early to avoid an error

joehoyle committed:
Copding standards

chadaort committed:
adding access to ga client and settings

joehoyle committed:
Merge pull request #12 from snopesdotcom/access-adj

adding access to ga client and settings

kevinlangleyjr committed:
Check if post_id exists in the array of top posts and add to the views count if so otherwise, set count.

kevinlangleyjr committed:
Merge pull request #13 from humanmade/fix_count_multiple_post_urls

Check if post_id exists in the array of top posts and add to the view…

rmccue commented:
@kevinlangleyjr Looks like you want to take over maintaining this plugin 😉 Shall we merge this pull request, given everyone's using it?

mikeselander commented:
Ping on this :D 

I've got a client using this and it'd be great to get master updated to the latest, or declare it dead.

mikeselander commented on hmtp.widget.php:
```diff
@@ -0,0 +1,155 @@
+ 		$title = isset( $instance[ 'title' ] ) ? $instance[ 'title' ] : 'Most Popular';
+
+ 		$args = wp_parse_args(
+ 			(array) $instance['args'],
```
There is no check ensuring that this key exists and it is thusly throwing an notice:

```
[30-Oct-2018 18:45:00 UTC] PHP Notice:  Undefined index: args in /chassis/content/plugins/hm-top-posts/hmtp.widget.php on line 74
[30-Oct-2018 18:45:00 UTC] PHP Stack trace:
[30-Oct-2018 18:45:00 UTC] PHP   1. {main}() /chassis/wordpress/wp-admin/widgets.php:0
[30-Oct-2018 18:45:00 UTC] PHP   2. wp_list_widgets() /chassis/wordpress/wp-admin/widgets.php:401
[30-Oct-2018 18:45:00 UTC] PHP   3. wp_widget_control() /chassis/wordpress/wp-admin/includes/widgets.php:48
[30-Oct-2018 18:45:00 UTC] PHP   4. HMTP_Widget->form_callback() /chassis/wordpress/wp-admin/includes/widgets.php:243
[30-Oct-2018 18:45:00 UTC] PHP   5. HMTP_Widget->form() /chassis/wordpress/wp-includes/class-wp-widget.php:507
[30-Oct-2018 18:45:17 UTC] PHP Notice:  Undefined index: args in /chassis/content/plugins/hm-top-posts/hmtp.widget.php on line 74
[30-Oct-2018 18:45:17 UTC] PHP Stack trace:
[30-Oct-2018 18:45:17 UTC] PHP   1. {main}() /chassis/wordpress/wp-admin/widgets.php:0
[30-Oct-2018 18:45:17 UTC] PHP   2. wp_list_widgets() /chassis/wordpress/wp-admin/widgets.php:401
[30-Oct-2018 18:45:17 UTC] PHP   3. wp_widget_control() /chassis/wordpress/wp-admin/includes/widgets.php:48
[30-Oct-2018 18:45:17 UTC] PHP   4. HMTP_Widget->form_callback() /chassis/wordpress/wp-admin/includes/widgets.php:243
[30-Oct-2018 18:45:17 UTC] PHP   5. HMTP_Widget->form() /chassis/wordpress/wp-includes/class-wp-widget.php:507
```

mikeselander reviewed (commented):
Could we possibly use Composer to pull in the GAPI client code? Would make it easier to maintain and much lighter.

goldenapples committed:
Only handle GA authentication on the defined redirect URI

This plugin was trying to perform oAuth authentication with Google
anytime it received a URL containing the query string '?code='. In
addition to breaking any other oauth integrations on a site, this
seems like just a bad idea in general.

This commit checks to make sure that the oauth handshake only happens on
the page defined as the ga_redirect_uri in the plugin settings -
ideally, this would be an admin page to prevent abuse.

goldenapples committed:
Merge pull request #15 from humanmade/check-redirect-uri

Only handle GA authentication on the defined redirect URI

kadamwhite committed:
Do not use curly-brace for string offset access

kadamwhite committed:
Pass glue and pieces in correct order when using implode()

kadamwhite committed:
Merge pull request #16 from humanmade/fix-php-7-4-warnings

Fix PHP 7.4 warnings
